rts/Game/Camera.cpp

```diff
@@ -774,22 +775,30 @@ float3 CCamera::GetMoveVectorFromState(bool fromKeyState) const
 	return v;
 }
 
-float3 CCamera::PointToMaxUnitAltitude(const float3& point, const float rayLength, const float maxAltitude) const
+bool CCamera::TracePointToMaxAltitude(const float3& point, const float rayLength, const float maxAltitude, float3& result) const
```

Comment:
Perhaps this could return `std::optional <float3>`?

Reply:
done